ui/selectors/transactions.js

```diff
@@ -596,3 +600,21 @@ export function hasTransactionPendingApprovals(state) {
     TRANSACTION_APPROVAL_TYPES.some((type) => hasPendingApprovals(state, type))
   );
 }
+
+const ConfirmationApprovalTypes = [
+  ApprovalType.EthSign,
+  ApprovalType.PersonalSign,
+  ApprovalType.EthSignTypedData,
+  ApprovalType.Transaction,
+];
+
+export function getPendingConfirmations(state) {
+  return getPendingApprovals(state).filter(({ type }) =>
+    ConfirmationApprovalTypes.includes(type),
+  );
+}
+
+export function getLatestPendingConfirmation(state) {
+  const pendingConfirmations = getPendingConfirmations(state);
+  return pendingConfirmations.sort((p1, p2) => p1.time < p2.time)[0];
+}
```

Comment:
Q: why create confirmation functions here rather than in `ui/selectors/confirm.ts`?

Reply:
Yep I fixed this already in following implementation PRs.